README.md

```diff
@@ -2,54 +2,196 @@
 
 [![Version][version-svg]][package-url] [![Build Status][travis-svg]][travis-url] [![License][license-image]][license-url] [![Downloads][downloads-image]][downloads-url]
 
-This is [Algolia](https://www.algolia.com/)'s [ESLint](http://eslint.org/) configuration.
+This is [Algolia](https://www.algolia.com/)'s linting and formatting of
+JavaScript projects (Vanilla, React, Vue) repository.
+
+It explains to you how to setup your project to use it and never
+worry again about indentation, curly spaces, let vs const etc...
+
+Just **focus** on coding.
 
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 **Table of Contents**
 
-- [Linting (ESLint)](#linting-eslint)
-- [Auto formatting staged files](#auto-formatting-staged-files)
+- [Setup your project](#setup-your-project)
+  - [Base requirements](#base-requirements)
+  - [Vanilla](#vanilla)
+  - [React](#react)
+  - [Flow](#flow)
+  - [Vue](#vue)
+- [Existing codebase setup](#existing-codebase-setup)
+- [Automatically fixing errors](#automatically-fixing-errors)
+  - [In your editor](#in-your-editor)
+  - [Before committing](#before-committing)
 - [Ignoring files](#ignoring-files)
 - [Test](#test)
 - [Release](#release)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## Linting (ESLint)
+## Setup your project
+
+### Base requirements
+
+```sh
+yarn add \
+eslint babel-eslint prettier \
+eslint-config-algolia eslint-config-prettier \
+eslint-plugin-import eslint-plugin-jest eslint-plugin-prettier \
+--dev
+```
+
+### Vanilla
+
+**.eslintrc.js**
+```js
+module.exports = {
+  extends: 'algolia'
+};
+```
+
+**package.json**
+```json
+{
+  "scripts": {
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
```

Comment:
True, at the same time next time we change the linting command might be when we have flying cars from Elon. Will change stil :)